cpp/include/cudf/detail/device_scalar.hpp

```diff
@@ -54,15 +54,15 @@ class device_scalar : public rmm::device_scalar<T> {
   explicit device_scalar(
     rmm::cuda_stream_view stream,
     rmm::device_async_resource_ref mr = cudf::get_current_device_resource_ref())
-    : rmm::device_scalar<T>(stream, mr), bounce_buffer{make_host_vector<T>(1, stream)}
+    : rmm::device_scalar<T>(stream, mr), bounce_buffer{make_pinned_vector<T>(1, stream)}
```

Comment:
I think we should change the default value of `allocate_host_as_pinned_threshold` instead of hard-coding the use of pinned memory here. We do need performance results to make either change. Happy to discuss :)